app/controllers/articles_controller.rb

```diff
@@ -30,23 +29,36 @@ def edit
 
   def update
     @article = Article.find(params[:id])
-
-    if @article.update(article_params)
-      redirect_to @article
+    if @article.user_id == current_user.id
+      if @article.update(article_params)
+        redirect_to @article
+      else
+        render :edit
+      end
     else
-      render :edit
+      redirect_to @article
     end
   end
 
   def destroy
     @article = Article.find(params[:id])
+    if @article.user_id == current_user.id
```

Comment:
@idprietop could you please explain this and also fix the code indentation, it is really hard to understand!

Reply:
In this line of code I am making sure that when a user is going to delete an article, first validate if this article belongs to the user who logged in.

Reply:
@idprietop ok, good!